### PR TITLE
Edit messages

### PR DESCRIPTION
**Pull request by cammellos**
Adds editing of messages.

I had to rebuild the state of input mentions, @rasom mentioned I could re-use some of the existing codebase, but I had already written the code, tried a bit but not too hard, if that's ok I'd rather try in a separate PR as we probably want to move to something else.

The fiddly bit is to have the input refresh, as we use normal atoms, so I had to increment the key after setting the text.

@Serhy @churik e2e tests needs to be updated as the design for replying to a message has changed (we don't show any information, just `Replaying to message`)
status: ready

### CODE REVIEW

**Comment by jakubgs** [comment]
Upgraded Go on `macos-03` from `1.13.4` to `1.15.13` in status-im/infra-role-golang@c49c4a2.
Test build: https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-12231/4/
@vkjr is having the same issue in #12242.

**Comment by jakubgs** [comment]
Nope, looks like the same thing happens:
```
14:02:32  # github.com/status-im/status-go/vendor/golang.org/x/sys/unix
14:02:32  ../../go/src/github.com/status-im/status-go/vendor/golang.org/x/sys/unix/syscall_darwin.go:383:45: undefined: Xucred
14:02:32  ../../go/src/github.com/status-im/status-go/vendor/golang.org/x/sys/unix/syscall_darwin.go:384:11: undefined: Xucred
14:02:32  ../../go/src/github.com/status-im/status-go/vendor/golang.org/x/sys/unix/syscall_darwin.go:385:21: undefined: SizeofXucred
```

**Comment by jakubgs** [comment]
Since `1.15` [drops support for ARM32 builds on iOS](https://golang.org/doc/go1.15#darwin) I've downgraded go to `1.14.15`: https://github.com/status-im/infra-role-golang/commit/d0892008
https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-12231/5/

**Comment by cammellos** [comment]
@vkjr @jakubgs I fixed it in this branch,
basically `go-waku` upgraded to a too recent version of `sys`.
I downgraded it to the release branch for `1.15` which does not include those changes, updated status-go and pointed this PR to that branch.
It's now building. 
I will run e2e and we can see how to get it in develop (probably as a separate PR, but we need to merge the go-waku changes first)
cc @siphiuel @richard-ramos

File: src/status_im/navigation/core.cljs [hunk 1]
```diff
@@ -67,7 +67,6 @@
     (if @dissmissing
       (reset! dissmissing comp)
       (do
-        (println "SHOW MODAL" comp)
```
Comment (flexsurfer) [inline src/status_im/navigation/core.cljs]: yeah sorry about that, will be fixed in my PR

File: ci/Jenkinsfile.android [hunk 2]
```diff
@@ -6,7 +6,7 @@ pipeline {
   options {
     timestamps()
     /* Prevent Jenkins jobs from running forever */
-    timeout(time: 20, unit: 'MINUTES')
+    timeout(time: 30, unit: 'MINUTES')
```
Comment (flexsurfer) [inline ci/Jenkinsfile.android:9]: why do we need this change ?

**Review by flexsurfer — APPROVED** [review]

File: ci/Jenkinsfile.android (hunk 2 above)
Comment (cammellos) [reply ci/Jenkinsfile.android:9]: The build kept timing out, we discussed it with @jakubgs , we can set it to 25 minutes

**Comment by Serhy** [comment]
@cammellos could you rebase against latest develop please? (RNN update merged recently)

**Comment by churik** [comment]
@cammellos 
Can we rebase it aging to avoid conflict in go (sorry)?

**Comment by churik** [comment]
@cammellos AMAZING WORK!!! still testing but it is supercool.

Several issues:
### ISSUE 1 [verified, no option to edit]: Missing translate when editing emoji message [x]

IMAGE 2021-06-24 14:25:11

### ISSUE 2 [NOT FIXED]: message from activity center disappears after editing

**Steps:**
1) User A: start message with User B
2) User B: pay attention that unread messages indicator is shown on activity centre
3) User A: edit message
4) User B: check activity centre

**Expected result:** unread messages indicator is shown, can open activity center and see new messages

**Actual result:**  unread messages indicator is shown, but no messages in Activity centre

**OS:** IOS, Android

### ISSUE 3 [VERIFIED]: edited messages are shown as new on home view [x]

**Steps:**
1) User A and User B are contacts
2) User A sends message to User B, when he is on home view
3) User A edit message

**Expected result:** 1 unread message

**Actual result:** 2 unread messages

**OS:** IOS, Android

### ISSUE 4 [VERIFIED]: error on long tap on emoji message from other user [x]
IMAGE 2021-06-24 14:39:23

### ISSUE 5 [VERIFIED]: when dismiss editing "Send"  button is active (looks like message itself remains in input, but it is not shown) [x]
**Steps:**
1) send message 
2) long tap on message > edit
3) cross icon

**Expected result:**
chat input looks like:
image

**Actual result:**
Monosnap 2021-06-24 14-48-35



### ISSUE 6(minor, not sure worth to fix): when editing message with mentions if move cursor inside mention i

**Steps:**
1) send message with mention
2) long tap on message > edit
3) move cursor inside mention 

**Expected result:**
nothing is changed 
**Actual result:**
IMAGE 2021-06-24 14:38:22

**OS:** IOS, Android

### ISSUE 7(question) [VERIFIED]: isn't cross icon too white?
photo_2021-06-24 14 42 14 jpeg 2021-06-24 14-43-24

**Comment by churik** [comment]
### ISSUE 8 [verified]: on each edited message PN is coming (Android only)

**Steps:**
User A and B are contacts
1) User A: send message 
2) User B (Android, PN is on): put app in background
3) User A: edit initial message

**Expected result:** no PN (or PN that message is edited???)

**Actual result:** PN with old text + PN with new text

**OS:** IOS, Android

### ISSUE 9 [verified]: send commands / images are available in editing mode [x]

Monosnap 2021-06-24 15-26-39

### ISSUE 10 [verified]: edited messages are not shown in group chat for recipient

**Steps:**
User A and B are in same group chat
1) User A: send message to group chat, edit it
2) User B: check group chat

**Expected result:** message is edited for both User A and B

**Actual result:** message is not changed for user B

**OS:** IOS, Android

### ISSUE 11 [verified]: error at opening 1-1 chat after editing to message with line break char
User A and B are contacts
1) User A: send message 
2) User B (Android, PN is on): put app in background
3) User A: Long tap > edit
4) User A: delete all in chat input and click enter

IMAGE 2021-06-24 16:22:49
Can't open 1-1 with this user anymore

**Comment by Serhy** [comment]
@cammellos is it possible in scope of this PR to add 'Edit Timeline posts' feature as well? User statuses also would be amazing to have ability to edit...

**Comment by cammellos** [comment]
@Serhy I'll see, I'll start by fixing the issues, if it goes well I can add it on this PR, if it takes too long probably we'll add it in a separate PR if that's ok

**Comment by cammellos** [comment]
@churik all the issues but 6 should be fixed (that one I'd fix separately).
Thanks for testing!

**Comment by Serhy** [comment]
> @Serhy I'll see, I'll start by fixing the issues, if it goes well I can add it on this PR, if it takes too long probably we'll add it in a separate PR if that's ok

OK to add in separate PR, yes (just thought it's low handing feature which we may include along the way)

**Comment by churik** [comment]
@cammellos can't build IOS, need it for testing

**Comment by cammellos** [comment]
@churik it built successfully now

**Comment by churik** [comment]
@cammellos 
Thank you! all issues, apart for 2, are fixed.
Issue 2 from https://github.com/status-im/status-react/pull/12231#issuecomment-867610434 is reproducible with same steps on builds with https://github.com/status-im/status-react/commit/60ea9682df5bce0e51e4b9039111b798db4398d9 commit

**Comment by cammellos** [comment]
@churik ah sorry, I missed it, I will take a look at it

**Comment by churik** [comment]
Fixes https://github.com/status-im/status-react/issues/12218

**Comment by churik** [comment]
All above issues were fixed, editing status will be added in separate issue.

Thank you @cammellos

**Pull request merged**